geode-protobuf/src/test/java/org/apache/geode/internal/protocol/protobuf/v1/DisconnectClientIntegrationTest.java

```diff
@@ -0,0 +1,138 @@
+import org.apache.geode.cache.RegionFactory;
+import org.apache.geode.cache.server.CacheServer;
+import org.apache.geode.distributed.ConfigurationProperties;
+import org.apache.geode.internal.AvailablePortHelper;
+import org.apache.geode.management.internal.security.ResourceConstants;
+import org.apache.geode.security.SecurityManager;
+import org.apache.geode.test.junit.categories.IntegrationTest;
+
+@Category(IntegrationTest.class)
+public class DisconnectClientIntegrationTest {
+  public static final String SECURITY_PRINCIPAL = "principle";
+  private Socket socket;
+  private Cache cache;
+  private SecurityManager securityManager;
+
+  @Rule
+  public RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
+
+  @Before
+  public void setUp() throws Exception {
+    CacheFactory cacheFactory = new CacheFactory(new Properties());
+    cacheFactory.set(ConfigurationProperties.MCAST_PORT, "0");
+    cacheFactory.set(ConfigurationProperties.ENABLE_CLUSTER_CONFIGURATION, "false");
+    cacheFactory.set(ConfigurationProperties.USE_CLUSTER_CONFIGURATION, "false");
+
+    securityManager = mock(SecurityManager.class);
+    cacheFactory.setSecurityManager(securityManager);
+    when(securityManager.authenticate(any())).thenReturn(SECURITY_PRINCIPAL);
+    when(securityManager.authorize(eq(SECURITY_PRINCIPAL), any())).thenReturn(true);
+
+    cache = cacheFactory.create();
+
+    CacheServer cacheServer = cache.addCacheServer();
+    int cacheServerPort = AvailablePortHelper.getRandomAvailableTCPPort();
+    cacheServer.setPort(cacheServerPort);
+    cacheServer.start();
+
+    RegionFactory<Object, Object> regionFactory = cache.createRegionFactory();
+    regionFactory.setDataPolicy(DataPolicy.PARTITION);
+
+
+    System.setProperty("geode.feature-protobuf-protocol", "true");
+
+    socket = new Socket("localhost", cacheServerPort);
+
+    Awaitility.await().atMost(5, TimeUnit.SECONDS).until(socket::isConnected);
+
+    MessageUtil.performAndVerifyHandshake(socket);
+  }
+
+  @After
+  public void tearDown() {
+    cache.close();
+    try {
+      socket.close();
+    } catch (IOException ignore) {
+      // NOP
+    }
+  }
+
+  @Test
+  public void disconnectsFromServer() throws Exception {
+    authenticateWithServer();
+
+    final ClientProtocol.Message requestMessage = createRequestMessageBuilder(
+        ConnectionAPI.DisconnectClientRequest.newBuilder().setReason("Normal termination")).build();
+
+    final ClientProtocol.Message responseMessage = writeMessage(requestMessage);
+    assertEquals(responseMessage.toString(),
+        ClientProtocol.Message.MessageTypeCase.DISCONNECTCLIENTRESPONSE,
+        responseMessage.getMessageTypeCase());
+    final ConnectionAPI.DisconnectClientResponse disconnectClientResponse =
+        responseMessage.getDisconnectClientResponse();
+    assertNotNull(disconnectClientResponse);
+    assertEquals(true, disconnectClientResponse.getAcknowledged());
```

Comment:
Use assertTrue (assuming you keep this field).

Reply:
'Tis gone.